builder/git/git.go

```diff
@@ -177,37 +201,31 @@ func createRepo(c cookoo.Context, repoPath, gitHome string) (bool, error) {
 	createLock.Lock()
 	defer createLock.Unlock()
 
-	if fi, err := os.Stat(repoPath); err == nil && fi.IsDir() {
-		// Nothing to do.
-		log.Infof(c, "Directory %s already exists.", repoPath)
-		return false, nil
-	} else if os.IsNotExist(err) {
-
-		log.Infof(c, "Creating new directory at %s", repoPath)
-		// Create directory
-		if err := os.MkdirAll(repoPath, 0755); err != nil {
-			log.Warnf(c, "Failed to create repository: %s", err)
-			return false, err
+	if fi, err := os.Stat(repoPath); err == nil {
+		if fi.IsDir() {
+			configPath := filepath.Join(repoPath, "config")
+			if _, cerr := os.Stat(configPath); cerr == nil {
+				log.Infof(c, "Directory '%s' already exists.", repoPath)
+				return true, nil
+			} else {
+				log.Warnf(c, "Path '%s' have not config file, remove it and recreate.", repoPath)
+				if err := os.RemoveAll(repoPath); err != nil {
+					return false, fmt.Errorf("Unable to remove path '%s': %s", repoPath, err)
+				}
+				return initRepo(repoPath, gitHome, c)
```

Comment:
There are many calls to `initRepo()` in the various branches of this conditional.  Move this to the end, after the conditional, since most paths need to result in this.  Paths that do not can just return early.